src/app/pages/storage/disks/disk-form/disk-form.component.ts

```diff
@@ -1,220 +1,133 @@
-        },
-      ],
-    },
-    {
-      name: helptext.fieldset_smartsed,
-      label: true,
-      class: 'general',
-      width: '100%',
-      config: [
-        {
-          type: 'checkbox',
-          name: 'togglesmart',
-          placeholder: helptext.disk_form_togglesmart_placeholder,
-          tooltip: helptext.disk_form_togglesmart_tooltip,
-        },
-        {
-          type: 'input',
-          name: 'smartoptions',
-          placeholder: helptext.disk_form_smartoptions_placeholder,
-          tooltip: helptext.disk_form_smartoptions_tooltip,
-        },
-        {
-          type: 'input',
-          name: 'passwd',
-          placeholder: helptext.disk_form_passwd_placeholder,
-          tooltip: helptext.disk_form_passwd_tooltip,
-          inputType: 'password',
-          value: '',
-          togglePw: true,
-          relation: [
-            {
-              action: RelationAction.Disable,
-              when: [
-                {
-                  name: 'clear_pw',
-                  value: true,
-                },
-              ],
-            },
-          ],
-        },
-        {
-          type: 'checkbox',
-          name: 'clear_pw',
-          placeholder: helptext.clear_pw.placeholder,
-          tooltip: helptext.clear_pw.tooltip,
-        },
-      ],
-    },
-    { name: 'divider', divider: true },
-  ];
-
-  title: string;
-
-  rowid: string;
+export class DiskFormComponent implements OnInit {
+  form = this.fb.group({
+    name: [''],
+    serial: [''],
+    description: [''],
+    critical: [null as number, [Validators.min(0)]],
+    difference: [null as number, [Validators.min(0)]],
+    informational: [null as number, [Validators.min(0)]],
+    hddstandby: [null as DiskStandby],
+    advpowermgmt: [null as DiskPowerLevel],
+    togglesmart: [false],
+    smartoptions: [],
+    passwd: [''],
+    clear_pw: [false],
+  });
+  readonly helptext = helptext;
+  readonly title = helptext.disk_form_title;
+  readonly hddstandbyOptions$ = of(helptext.disk_form_hddstandby_options);
+  readonly advpowermgmtOptions$ = of(this.translateOptions(this.helptext.disk_form_advpowermgmt_options));
+  isLoading = false;
+  existingDisk: Disk;
 
   constructor(
-    private router: Router,
-    protected ws: WebSocketService,
-    protected aroute: ActivatedRoute,
+    private translate: TranslateService,
+    private ws: WebSocketService,
+    private fb: FormBuilder,
+    private dialogService: DialogService,
+    private cdr: ChangeDetectorRef,
+    private errorHandler: FormErrorHandlerService,
+    private slideInService: IxSlideInService,
   ) {
   }
 
-  resourceTransformIncomingRestData(data: Disk): Disk {
-    const transformed = { ...data };
-    delete transformed.passwd;
-    return transformed;
+  ngOnInit(): void {
+    this.clearPasswordField();
+  }
+
+  setFormDisk(disk: Disk): void {
+    this.existingDisk = disk;
+    const transformedData = { ...disk };
+    delete transformedData.passwd;
```

Comment:
I don't think we need to remove current password. It provides good indication that SED password has been set.